twitch.py

```diff
@@ -828,19 +828,40 @@ async def _run(self):
                 exclude = self.settings.exclude
                 priority = self.settings.priority
                 priority_only = self.settings.priority_only
+                priority_by_time = self.settings.priority_by_time
                 next_hour = datetime.now(timezone.utc) + timedelta(hours=1)
-                for campaign in self.inventory:
+                campaigns = self.inventory
+                campaigns.sort(key=lambda c: c.ends_at, reverse=False) # lets us give priority to campaigns ending soonest later on
+                max_priority = len(campaigns)
+                max_wanted_priority = max_priority + len(priorities)
+                for campaign in campaigns:
                     game = campaign.game
                     if (
-                        game not in self.wanted_games  # isn't already there
-                        and game.name not in exclude  # and isn't excluded
+                        game not in self.wanted_games # isn't already there
+                        and game.name not in exclude # and isn't excluded
                         # and isn't excluded by priority_only
                         and (not priority_only or game.name in priority)
                         # and can be progressed within the next hour
                         and campaign.can_earn_within(next_hour)
                     ):
-                        # non-excluded games with no priority are placed last, below priority ones
-                        self.wanted_games[game] = priorities.get(game.name, 0)
+                        # get users priority preference
+                        game_priority = priorities.get(game.name, 0)
+
+                        if (game_priority):
+                            if (not priority_by_time):
+                                # keep games ordered by priority list
+                                #    max_wanted_priority does not change in this case, consider it "0"
+                                self.wanted_games[game] = game_priority + max_wanted_priority
```

Comment:
Because `max_wanted_priority` is not decremented if not sorting by time, this number can be ignored for the sake of sorting. 
It's only added here to give all prioritized games a higher priority number than non-priority games.